azure-proto-core/Resources/Sku.cs

```diff
@@ -46,11 +51,16 @@ public bool Equals(Sku other)
                 return false;
             }
 
-            if (object.Equals(Name, other.Name) &&
-                object.Equals(Family, other.Family) &&
-                object.Equals(Size, other.Size) &&
-                object.Equals(Tier, other.Tier) &&
-                object.Equals(Capacity, other.Capacity))
+            if (object.Equals(this, other))
```

Comment:
Can we make this explicit since since thats all its really doing?  There are 4 instances of this.
```suggestion
            if (object.ReferenceEquals(this, other))
```